sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py

```diff
@@ -86,6 +86,7 @@ def __init__(self, base_url: str, credential: AsyncTokenCredential, **kwargs: An
             credential_policy = AppConfigRequestsCredentialsPolicy(credential)  # type: ignore
 
         self._impl = AzureAppConfiguration(
+            credential,
```

Comment:
It is weird we need to provide both credential and credential policy.

Reply:
After investigation, I notice `credential_policy` is not need to pass in aad mode now, the new gernerated code can do the same thing: https://github.com/Azure/azure-sdk-for-python/blob/2ac043ba38fa6dc3e6097d9e70271965ec7c8b04/sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_generated/aio/_configuration.py#L69-L72

@iscai-msft Do you know why there's a `*` for parameter credential_scopes?  It potentially is a bug in autorest. I got this error with it: 
```text
E       azure.core.exceptions.ClientAuthenticationError: Microsoft Entra ID error '(invalid_scope) AADSTS70011: The provided request must include a 'scope' input parameter. The provided value for the input parameter 'scope' is not valid. The scope h t t p s : / / y a l l a p p c o n f i g t e s t s - a z c o n f i g - p y t h o n . a z c o n f i g . i o / . d e f a u l t is not valid. Trace ID: 5764f775-ced1-4c23-ab35-3e0259c83b00 Correlation ID: 6832454b-1d75-47c6-994b-5ea247dcf311 Timestamp: 2023-10-26 18:45:26Z'
E       Content: {"error":"invalid_scope","error_description":"AADSTS70011: The provided request must include a 'scope' input parameter. The provided value for the input parameter 'scope' is not valid. The scope h t t p s : / / y a l l a p p c o n f i g t e s t s - a z c o n f i g - p y t h o n . a z c o n f i g . i o / . d e f a u l t is not valid. Trace ID: 5764f775-ced1-4c23-ab35-3e0259c83b00 Correlation ID: 6832454b-1d75-47c6-994b-5ea247dcf311 Timestamp: 2023-10-26 18:45:26Z","error_codes":[70011],"timestamp":"2023-10-26 18:45:26Z","trace_id":"5764f775-ced1-4c23-ab35-3e0259c83b00","correlation_id":"6832454b-1d75-47c6-994b-5ea247dcf311"}
```
Also tried without `*`, the same test works well.

Reply:
credential scopes should be a list, are you passing in a string?

Reply:
Ah, I'm passing in a string